packages/core-plugin-api/src/analytics/AnalyticsContext.tsx

```diff
@@ -14,23 +14,40 @@
  * limitations under the License.
  */
 
-import React, { createContext, ReactNode, useContext } from 'react';
+import {
+  createVersionedContext,
+  createVersionedValueMap,
+} from '@backstage/version-bridge';
+import React, { ReactNode, useContext } from 'react';
 import { AnalyticsContextValue } from './types';
 
-// todo(iamEAP): Manage this using a version bridge.
-const AnalyticsReactContext = createContext<AnalyticsContextValue>({
-  routeRef: 'unknown',
-  pluginId: 'root',
-  extension: 'App',
-});
+const AnalyticsReactContext =
+  createVersionedContext<{ 1: AnalyticsContextValue }>('analytics-context');
 
 /**
  * A "private" (to this package) hook that enables context inheritance and a
  * way to read Analytics Context values at event capture-time.
  * @private
  */
-export const useAnalyticsContext = () => {
-  return useContext(AnalyticsReactContext);
+export const useAnalyticsContext = (): AnalyticsContextValue => {
+  const theContext = useContext(AnalyticsReactContext);
+
+  // Provide a default value if no value exists.
+  if (theContext === undefined) {
+    return {
+      routeRef: 'unknown',
+      pluginId: 'root',
+      extension: 'App',
+    };
+  }
+
+  // This should probably never happen, but check for it.
```

Comment:
It can happen in the future if you have an old version of the core API consuming a future API version that no longer provider version 1 of the API